gitpod-io/gitpod
# Handle an expired admin user link

## Description

Adds an error page to the admin login flow. I added a layer above our `<App/>` that provides a way to add routes for things like addressable error pages. I've added `/error/expired-ots` for this scenario, but anything under `/error/*` will resolve to this new routing layer, and render a default error page if we don't have an explicit route for it yet. This will let us add more addressable error pages in the future if we need. The primary intent of these is for scenarios where the api layer needs to redirect somewhere in the case of an error, and let us render clientside to accomplish that. No external data is loaded from the client for this routing layer, such as a user/orgs.

![image](https://github.com/gitpod-io/gitpod/assets/367275/dac33493-01cb-4360-b645-f5eada34cd21)

## Related Issue(s)

Fixes WEB-286

## How to test

* You can manually view the error page by visiting `/error/expired-ots`
* Visiting `/error/foo` will render a default error page, mostly to just have a catch-all for any undefined routes underneath the `/error/*` path.
* Simulate a bad ots by visiting `/api/login/ots/admin/badtoken` - it should redirect to the `/error/expired-ots` error page

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-web-21aa1d24e1f</li>
	<li><b>🔗 URL</b> - <a href="https://brad-web-21aa1d24e1f.preview.gitpod-dev.com/workspaces" target="_blank">brad-web-21aa1d24e1f.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
